src/Identity/Core/src/LoggingExtensions.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{ 
+    internal static class LoggingExtensions
+    {
+        private static readonly Action<ILogger, Exception> _invalidExpirationTime;
+        private static readonly Action<ILogger, Exception> _userIdsNotEquals;
+        private static readonly Action<ILogger, Exception> _purposeNotEquals;
+        private static readonly Action<ILogger, Exception> _unexpectedEndOfInput;
+        private static readonly Action<ILogger, Exception> _securityStampNotEquals;
+        private static readonly Action<ILogger, Exception> _securityStampIsNotEmpty;
+        private static readonly Action<ILogger, Exception> _unhandledException;
+
+        static LoggingExtensions()
+        {
+            _invalidExpirationTime = LoggerMessage.Define(
+                eventId: new EventId(0, "InvalidExpirationTime"),
+                logLevel: LogLevel.Information,
+                formatString: "ValidateAsync failed: the expiration time is invalid.");
+
+            _userIdsNotEquals = LoggerMessage.Define(
+                eventId: new EventId(1, "UserIdsNotEquals"),
+                logLevel: LogLevel.Debug,
+                formatString: "ValidateAsync failed: the expected and actual UserId aren't equal");
```

Comment:
nit: "ValidateAsync failed: did not find expected UserId."  Note there's also a period missing from this one